### PR TITLE
Super repository refactor + nice half-baked generator script

### PR DESCRIPTION
**Pull request by mateusfccp**
This solves #10 in a different way than the proposed one.

I made a CL script to generate the readmes, both the main one and the specific ones. This script has been hastened-made, so it needs a lot of polishment that I am going to do later. This scripts works as long as you respect the following:
* Execute the `(main)` function from within `builder` folder.
* Each individual cat has to be a folder within `cats` that has its name (it will be used to generate the readmes). Conflicts will fail.
* Currently, only supports (manually whitelisted) jpg, jpeg, bmp, png and gif extensions. Don't know if it is need to add different ones, tho.

I am going to make some polishment and changes to improve the overall experience and flexibility, and turn the script in a compilable and CLI form. I am also thinking on supporting individual templates (like the `template.md`) or some other mechanism so contributors may be able to customize and give info about their cats.

@ryukinix Please, consider labeling this with `hacktoberfest-accepted`.

### CODE REVIEW

**Comment by mateusfccp** [comment]
@ryukinix Maybe you will want me to squash the commits?

**Comment by ryukinix** [comment]
> @ryukinix Maybe you will want me to squash the commits?

No problem! I can squash on merge.

File: builder/single-page.lisp [hunk 1]
```diff
@@ -0,0 +1 @@
+(defun
```
Comment (ryukinix) [inline builder/single-page.lisp]: What is that?

File: builder/main.lisp [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  "Returns a new string with substring replaced with replacement."
+  (let ((needle-position (search substring string)))
+    (if (null needle-position)
+        string
+        (let* ((needle-length (length substring))
+               (before (subseq string 0 needle-position))
+               (after (subseq string (+ needle-position needle-length))))
+          (concatenate 'string before replacement after)))))
+
+(defconstant image-extensions '("jpg" "jpeg" "bmp" "png" "gif"))
```
Comment (ryukinix) [inline builder/main.lisp]: Put constants on top of the file. Make it easier to read.

File: builder/main.lisp [hunk 3]
```diff
@@ -0,0 +1,81 @@
+;;; Core
+
+(defun main ()
```
Comment (ryukinix) [inline builder/main.lisp:3]: It is more a common pattern to define the main function in the end of the file. Make it the code reading more linear

File: builder/main.lisp [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  "Return only the pathname's folder as string instead of the entire path."
+  (first (last (pathname-directory pathname))))
+
+;; Generators
+
+(defun generate-table (cats-names)
+  (str-replace "<table>$1</table>" "$1" (generate-table-rows cats-names)))
+
+(defun generate-table-cell (cat-name)
+  (str-replace (str-replace (str-replace "<td align=\"center\"><a href=\"./cats/$3\">$1<strong>$2</strong></a></td>" "$1" (generate-img (get-cat-profile-picture cat-name))) "$2" cat-name) "$3" cat-name))
```
Comment (ryukinix) [inline builder/main.lisp]: That line is difficult to read due to size of line. Can you indent it?

File: builder/main.lisp [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+(defun generate-table-rows (cats-names)
```
Comment (ryukinix) [inline builder/main.lisp]: You are using later a cdddr but the input is not clear. Give a example of cats-names structure on the defun docstring

**Review by ryukinix — CHANGES_REQUESTED** [review]
Very nice @mateusfccp! I didn't expected that. Nice to see you coding in Lisp!

I requested a few changes to improve our code. Take a look!

File: builder/single-page.lisp (hunk 1 above)
Comment (mateusfccp) [reply builder/single-page.lisp]: Some debris from an earlier idea.

File: builder/main.lisp (hunk 2 above)
Comment (mateusfccp) [reply builder/main.lisp]: I am splitting the three "sections" of this file and putting the `(defconst` in the top of the `utils.lisp` file.

File: builder/main.lisp (hunk 5 above)
Comment (mateusfccp) [reply builder/main.lisp]: I removed `cdddr` and inserted a docstring. Is it clearer now?

Comment (ryukinix) [reply builder/main.lisp]: Yes, it is.

**Review by ryukinix — APPROVED** [review]
LGTM

**Comment by mateusfccp** [comment]
@ryukinix You forgot to squash the PR :v

**Comment by ryukinix** [comment]
@mateusfccp shit! Sorry, I'll fix in the master manually.

**Comment by ryukinix** [comment]
Fixed @mateusfccp!

**Pull request merged**